File: app/src/main/java/com/example/ikastyle/UI/LoadoutRecyclerViewAdapter.java
package com.example.ikastyle.UI;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.ikastyle.Common.Const.ResourceIdMap;
import com.example.ikastyle.Common.Util;
import com.example.ikastyle.Entity.GearSet;
import com.example.ikastyle.R;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

import java.util.List;

public class LoadoutRecyclerViewAdapter extends RecyclerView.Adapter<LoadoutRecyclerViewAdapter.ViewHolder> {
    private final List<GearSet> gearSetList;
    private final View.OnClickListener onClickDeleteListener;

    static class ViewHolder extends RecyclerView.ViewHolder{
        private TextView loadoutNameView;
        private ImageView headGearView;
        private ImageView headMainView;
        private ImageView headSub1View;
        private ImageView headSub2View;
        private ImageView headSub3View;
        private ImageView clothingGearView;
        private ImageView clothingMainView;
        private ImageView clothingSub1View;
        private ImageView clothingSub2View;
        private ImageView clothingSub3View;
        private ImageView shoesGearView;
        private ImageView shoesMainView;
        private ImageView shoesSub1View;
        private ImageView shoesSub2View;
        private ImageView shoesSub3View;
        private FloatingActionButton analyseView;
        private FloatingActionButton editView;
        private LoadoutDeleteButton deleteView;

        public ViewHolder(View view){
            super(view);
            loadoutNameView = view.findViewById(R.id.textView_loadoutName);
            headGearView = view.findViewById(R.id.imageView_headgear);
            headMainView = view.findViewById(R.id.imageView_head_main);
            headSub1View = view.findViewById(R.id.imageView_head_sub1);
            headSub2View = view.findViewById(R.id.imageView_head_sub2);
            headSub3View = view.findViewById(R.id.imageView_head_sub3);
            clothingGearView = view.findViewById(R.id.imageView_clothing);
            clothingMainView = view.findViewById(R.id.imageView_clothing_main);
            clothingSub1View = view.findViewById(R.id.imageView_clothing_sub1);
            clothingSub2View = view.findViewById(R.id.imageView_clothing_sub2);
            clothingSub3View = view.findViewById(R.id.imageView_clothing_sub3);
            shoesGearView = view.findViewById(R.id.imageView_shoes);
            shoesMainView = view.findViewById(R.id.imageView_shoes_main);
            shoesSub1View = view.findViewById(R.id.imageView_shoes_sub1);
            shoesSub2View = view.findViewById(R.id.imageView_shoes_sub2);
            shoesSub3View = view.findViewById(R.id.imageView_shoes_sub3);
            analyseView = view.findViewById(R.id.floatingActionButton_analyse);
            editView = view.findViewById(R.id.floatingActionButton_edit);
            deleteView = view.findViewById(R.id.floatingActionButton_delete);
        }
    }

    public LoadoutRecyclerViewAdapter(List<GearSet> gearSetList, View.OnClickListener onClickDeleteListener){
        this.gearSetList = gearSetList;
        this.onClickDeleteListener = onClickDeleteListener;
    }

    @Override
    @NonNull
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType){
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.cardview_layout_store, parent, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position){
        GearSet gearSet = gearSetList.get(position);

        holder.loadoutNameView.setText(gearSet.name);
        holder.headGearView.setImageResource(Util.getHeadGearResourceId(gearSet.headGearId));
        holder.headMainView.setImageResource(Util.getGearPowerResourceId(gearSet.headMain));
        holder.headSub1View.setImageResource(Util.getGearPowerResourceId(gearSet.headSub1));
        holder.headSub2View.setImageResource(Util.getGearPowerResourceId(gearSet.headSub2));
        holder.headSub3View.setImageResource(Util.getGearPowerResourceId(gearSet.headSub3));
        holder.clothingGearView.setImageResource(Util.getClothingResourceId(gearSet.clothingGearId));
        holder.clothingMainView.setImageResource(Util.getGearPowerResourceId(gearSet.clothingMain));
        holder.clothingSub1View.setImageResource(Util.getGearPowerResourceId(gearSet.clothingSub1));
        holder.clothingSub2View.setImageResource(Util.getGearPowerResourceId(gearSet.clothingSub2));
        holder.clothingSub3View.setImageResource(Util.getGearPowerResourceId(gearSet.clothingSub3));
        holder.shoesGearView.setImageResource(Util.gerShoesResourceId(gearSet.shoesGearId));
        holder.shoesMainView.setImageResource(Util.getGearPowerResourceId(gearSet.shoesMain));
        holder.shoesSub1View.setImageResource(Util.getGearPowerResourceId(gearSet.shoesSub1));
        holder.shoesSub2View.setImageResource(Util.getGearPowerResourceId(gearSet.shoesSub2));
        holder.shoesSub3View.setImageResource(Util.getGearPowerResourceId(gearSet.shoesSub3));
        holder.deleteView.setGearSet(gearSet);
        holder.deleteView.setOnClickListener(onClickDeleteListener);
    }

    @Override
    public int getItemCount(){
        return gearSetList.size();
    }
}

File: app/src/main/java/com/splatool/ikastyle/ui/LoadoutRecyclerViewAdapter.kt
package com.splatool.ikastyle.ui

import com.splatool.ikastyle.R
import androidx.recyclerview.widget.RecyclerView
import com.splatool.ikastyle.model.data.entity.Loadout
import android.view.*
import android.widget.*
import androidx.databinding.DataBindingUtil
import com.splatool.ikastyle.common.Util
import com.splatool.ikastyle.databinding.CardviewLayoutStoreBinding
import com.splatool.ikastyle.viewModel.StoreViewModel

class LoadoutRecyclerViewAdapter(private val storeViewModel: StoreViewModel) : RecyclerView.Adapter<LoadoutRecyclerViewAdapter.ViewHolder?>() {
    private var loadoutList : List<Loadout> = listOf()

    fun setLoadoutList(loadoutList : List<Loadout>){
        this.loadoutList = loadoutList

        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = DataBindingUtil.inflate(
            LayoutInflater.from(parent.context),
            R.layout.cardview_layout_store,
            parent,
            false
        ) as CardviewLayoutStoreBinding
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val loadout = loadoutList[position]
        holder.binding.loadout = loadout
        holder.binding.viewModel = storeViewModel

        // 画像の設定はDataBindingで出来なさそうなのでここで個別に設定する
        holder.binding.imageViewHeadgear.setImageResource(Util.getHeadGearResourceId(loadout.headGearId))
        holder.binding.imageViewHeadMain.setImageResource(Util.getGearPowerResourceId(loadout.headMain))
        holder.binding.imageViewHeadSub1.setImageResource(Util.getGearPowerResourceId(loadout.headSub1))
        holder.binding.imageViewHeadSub2.setImageResource(Util.getGearPowerResourceId(loadout.headSub2))
        holder.binding.imageViewHeadSub3.setImageResource(Util.getGearPowerResourceId(loadout.headSub3))
        holder.binding.imageViewClothing.setImageResource(Util.getClothingResourceId(loadout.clothingGearId))
        holder.binding.imageViewClothingMain.setImageResource(Util.getGearPowerResourceId(loadout.clothingMain))
        holder.binding.imageViewClothingSub1.setImageResource(Util.getGearPowerResourceId(loadout.clothingSub1))
        holder.binding.imageViewClothingSub2.setImageResource(Util.getGearPowerResourceId(loadout.clothingSub2))
        holder.binding.imageViewClothingSub3.setImageResource(Util.getGearPowerResourceId(loadout.clothingSub3))
        holder.binding.imageViewShoes.setImageResource(Util.getShoesResourceId(loadout.shoesGearId))
        holder.binding.imageViewShoesMain.setImageResource(Util.getGearPowerResourceId(loadout.shoesMain))
        holder.binding.imageViewShoesSub1.setImageResource(Util.getGearPowerResourceId(loadout.shoesSub1))
        holder.binding.imageViewShoesSub2.setImageResource(Util.getGearPowerResourceId(loadout.shoesSub2))
        holder.binding.imageViewShoesSub3.setImageResource(Util.getGearPowerResourceId(loadout.shoesSub3))

        holder.binding.executePendingBindings()
    }

    override fun getItemCount(): Int {
        return loadoutList.size
    }

    class ViewHolder(val binding : CardviewLayoutStoreBinding) : RecyclerView.ViewHolder(binding.root)
}